### ISSUE TITLE
AAX fails to build with deployment target 10.8 on Xcode 11

### ISSUE THREAD

**Issue by danra** [McMartin/FRUT#569]
Trying to build an AAX plugin using FRUT, on Xcode 11, with a deployment target of 10.8, I'm getting the following error when trying to build using the generate files.
`error: include path for stdlibc++ headers not found; pass ‘-std=libc++’`

Changing the deployment target to something higher seems to eliminate the issue.

I guess the following logic doesn't work correctly in my case:
https://github.com/McMartin/FRUT/blob/40e96f764e9139033cc1a3b3558ced4c81188a0d/cmake/Reprojucer.cmake#L2584

AFAIK `libc++` is compatible with 10.7 and up, so maybe we should just change 10.8 to 10.6 here? Though I suppose just doing this naively might cause issues with older Xcode versions which only have `libstdc++`.

**Comment by McMartin** [comment]
Here `Reprojucer.cmake` only attempts to reproduce what Projucer does. However, what Projucer does changed throughout the versions. 

In 4.2.0, it was:
```cpp
            const RelativePath aaxLibsFolder = RelativePath (owner.getAAXPathValue().toString(), RelativePath::projectFolder).getChildFile ("Libs");


            xcodeExtraLibrariesDebug.add   (aaxLibsFolder.getChildFile ("Debug/libAAXLibrary.a"));
            xcodeExtraLibrariesRelease.add (aaxLibsFolder.getChildFile ("Release/libAAXLibrary.a"));
```

Then in https://github.com/WeAreROLI/JUCE/commit/24d02528e034641d1c48fe1ffe1a568d71aa3f72 (a bit before 5.0.0), it became:
```cpp
                 auto aaxLibsFolder
                    = RelativePath (owner.getAAXPathValue().toString(), RelativePath::projectFolder)
                        .getChildFile ("Libs");


                String libraryPath (config.isDebug() ? "Debug/libAAXLibrary" : "Release/libAAXLibrary");
                libraryPath += (isUsingClangCppLibrary (config) ? "_libcpp.a" : ".a");


                extraLibs.add   (aaxLibsFolder.getChildFile (libraryPath));
```

And finally in https://github.com/WeAreROLI/JUCE/commit/c7c26fa9f033c196c48e743deb6ae16e524ff29a (between 5.3.1 and 5.3.2), it became:
```cpp
                 auto aaxLibsFolder
                    = RelativePath (owner.getAAXPathValue().toString(), RelativePath::projectFolder)
                        .getChildFile ("Libs");


                String libraryPath (config.isDebug() ? "Debug" : "Release");
                libraryPath += "/libAAXLibrary_libcpp.a";


                extraLibs.add   (aaxLibsFolder.getChildFile (libraryPath));
```

`Reprojucer.cmake` currently reproduces the second behavior, without respecting the `JUCE_VERSION` passed to `jucer_project_begin`. That's where the bug lies.

**Comment by McMartin** [comment]
@danra I pushed a branch with a tentative fix: [`fix-appending-_libcpp-to-AAX-libraries`](https://github.com/McMartin/FRUT/tree/fix-appending-_libcpp-to-AAX-libraries). Could you please give it a try and let me know whether that solves your problem? Thanks!

**Comment by danra** [comment]
With deployment target 10.8, I'm now getting
`warning: include path for stdlibc++ headers not found; pass '-stdlib=libc++' on the command line to use the libc++ standard library instead [-Wstdlibcxx-not-found]`

Followed by
```
...JUCE/modules/juce_core/system/juce_StandardHeader.h:46:10: fatal error: 'memory' file not found
#include <memory>
         ^~~~~~~~


** BUILD FAILED **
```

This is whether or not I'm building the AAX format - also getting this with just building standalone.

Not getting this behavior when I set the deployment target to 10.9.

**Comment by McMartin** [comment]
Since https://github.com/WeAreROLI/JUCE/commit/c7c26fa9f, Projucer always selects `libc++` as the standard library. I need to replicate that behavior in `Reprojucer.cmake`. I'll let you know once I'm done with it, I'm pretty sure I will create a new branch.

**Comment by McMartin** [comment]
Hi @danra,

Happy New Year!

I pushed a new branch to try to fix this issue completely: [`handle-JUCE-5.3.2-targeting-OS-X-10.7-minimum`](https://github.com/McMartin/FRUT/tree/handle-JUCE-5.3.2-targeting-OS-X-10.7-minimum). Could you please test it and let me know if I missed something? Thanks a lot for your help and your patience.

**Comment by danra** [comment]
It works great!

Thanks!
Dan
